### ISSUE TITLE
Neo4j 3.4.4: native string indexing fails for strings with length < 4096

### ISSUE THREAD

**Issue by cmorgner** [neo4j/neo4j#11993]
Neo4j version: 3.4.4
Operating system: (Kubuntu 18.04)
API/Driver: (Bolt Java Driver v1.6.1)

## Description
When creating a node with a large string property, the transaction fails even for strings with fewer characters than the documented limit. The transaction error message says `4096` characters while the documentation at https://neo4j.com/docs/operations-manual/current/performance/index-configuration/ says `4046` characters.

In my tests, the actual limit causing a tx failure was 4077 characters, with an error message different than expected. A length in the range of 4077 to 4095 characters causes the transaction to fail unexpectedly.

## Steps to reproduce
Run the following test class against a Neo4j 3.4.4 instance.

```java
public class IndexTest {

	public static void main(final String[] args) {

		try (final Driver driver = GraphDatabase.driver("bolt://localhost:7687",
						AuthTokens.basic("neo4j", "admin"),
						Config.build().withEncryptionLevel(Config.EncryptionLevel.NONE).toConfig()
			)) {

			try (final Session session = driver.session()) {

				try (final Transaction tx = session.beginTransaction()) {

					tx.run("CREATE INDEX ON :Test(largeString)");
					tx.success();
				}

				try (final Transaction tx = session.beginTransaction()) {

					final Map<String, Object> params = new HashMap<>();
					final Map<String, Object> data   = new HashMap<>();
					final StringBuilder buf          = new StringBuilder();

					for (int i=0; i<4090; i++) {
						buf.append("0");
					}

					data.put("largeString", buf.toString());
					params.put("data", data);

					tx.run("CREATE (n:Test { data }) RETURN n", params);

					tx.success();
				}
			}
		}
	}
}
```

## Expected behavior
Transaction success, no error message.

## Actual behavior
Transaction failure with the following error message.

    org.neo4j.driver.v1.exceptions.DatabaseException: Could not apply the transaction to the store after written to log

## Additional information
It seems that the check for the string length before indexing a property uses a threshold different from the one in the actual index.

**Comment by cmorgner** [comment]
Even worse: the error seems to be unrecoverable:

```
SCHWERWIEGEND: [0x72a6793c] Fatal error occurred in the pipeline
org.neo4j.driver.v1.exceptions.DatabaseException: The database has encountered a critical error, and needs to be restarted. Please see database logs for more details.
	at org.neo4j.driver.internal.util.ErrorUtil.newNeo4jError(ErrorUtil.java:66)
	at org.neo4j.driver.internal.async.inbound.InboundMessageDispatcher.handleFailur
```

**Comment by cmorgner** [comment]
Additional info: the erroneous index entry seems to cause more problems later when accessing the node:

```
SCHWERWIEGEND: [0xadb87392] Fatal error occurred in the pipeline
org.neo4j.driver.v1.exceptions.DatabaseException: org.neo4j.io.pagecache.CursorException: Read unreliable key, keySize=4082, keyValueSizeCap=4047, keyHasTombstone=false, offset=4084, pos=0 | GB+Tree[file:/tmp/structr-test-1532357003249/db/schema/index/lucene_native-2.0/219/string-1.0/index-219, layout:StringLayout[version:0.1, identifier:24016946018123776], generation:4/6]
	at org.neo4j.driver.internal.util.ErrorUtil.newNeo4jError(ErrorUtil.java:66)
```

**Comment by MishaDemianenko** [comment]
@cmorgner thx for reporting. 
@burqen can you please take a look?

**Comment by cmorgner** [comment]
Correction: the CursorException is caused by a value that did *not* cause the transaction to fail, e.g. 4074 or below.

**Comment by burqen** [comment]
@cmorgner I can confirm your observations. Thank you again for reporting. This will be addressed immediately.

**Comment by burqen** [comment]
The fix will be included in 3.4.6. Closing issue.
